src/semver/version.py

```diff
@@ -661,6 +661,37 @@ def isvalid(cls, version: str) -> bool:
         except ValueError:
             return False
 
+    def is_compatible(self, new: 'Version') -> bool:
+        """
+        Check if current version is compatible with new.
+
+        The algorithm checks:
+
+        * if the two majors are different, it's incompatible
+        * if the two majors are equal, but the minor of ``new``
+          is lower than self.minor, it's incompatible
+        * if both pre-releases are present and different,
+          it's incompatible
+        * otherwise it's compatible
+
+        The algorithm does *not* check patches.
+
+        :param new: the new version to check for compatibility
+        :return: True, if new is compatible with the old version,
+                 otherwise False
+        """
+        if not isinstance(new, type(self)):
+            raise TypeError(
+                f"Expected a Version type but got {type(new)}"
+            )
+        if self._major != new.major:
+            return False
+        if new.minor < self._minor:
+            return False
+        if self._prerelease != new.prerelease:
+            return False
+        return True
```

Comment:
```suggestion
        return self._major == new.major and new.minor > self._minor and self.prerelease == new.prerelease
```
Here is a suggestion for a slightly cleaner check (IMO). I wrote this in the Github UI so you may want to run black on it to format it properly since it is a rather long statement.

Reply:
Yes, I was thinking about the same. My approach was to make it obvious first and optimize later. :wink: 

However, when I add your code, I get these errors:

```
$ tox -e checks,py37
[...]
tests/test_semver.py ...............FFFF.......
[...]
______________________________________ [doctest] check-compatible-semver-version.rst _______________________________________
020 
021     # Two different majors:
022     >>> a = Version(1, 1, 1) # This is "A"
023     >>> b = Version(2, 0, 0) # This is "B"
024     >>> a.is_compatible(b)
025     False
026 
027     # The same two majors and minors:
028     >>> b = Version(1, 1, 0) # This is another "B"
029     >>> a.is_compatible(b)
Differences (ndiff with -expected +actual):
    - True
    + False

/home/tom/repos/GH/tomschr/python-semver/docs/usage/check-compatible-semver-version.rst:29: DocTestFailure
================================================= short test summary info ==================================================
FAILED tests/test_semver.py::test_should_succeed_compatible_match[old0-new0] - assert False
FAILED tests/test_semver.py::test_should_succeed_compatible_match[old1-new1] - assert False
FAILED tests/test_semver.py::test_should_succeed_compatible_match[old2-new2] - assert False
FAILED tests/test_semver.py::test_should_succeed_compatible_match[old3-new3] - AssertionError: assert False
FAILED docs/usage/check-compatible-semver-version.rst::check-compatible-semver-version.rst
============================================== 5 failed, 334 passed in 0.55s ===============================================
E
```

Reply:
Oops I didn't invert the minor condition properly 😆
```suggestion
        return self._major == new.major and new.minor >= self._minor and self.prerelease == new.prerelease
```

Reply:
Haha, no problem. :slightly_smiling_face: I've corrected it. Could you also answer the question in https://github.com/python-semver/python-semver/pull/368#issuecomment-1320319059 ? Many thanks!